share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorService.java

```diff
@@ -322,7 +322,7 @@ private CompletableFuture<Void> performRecordPruning(TopicPartition tp) {
                     return;
                 }
 
-                log.info("Pruning records in {} till offset {}.", tp, off);
+                log.debug("Pruning records in {} till offset {}.", tp, off);
```

Comment:
Just for my undersatnding: Why log at line 326 is in debug though there is an exception?

Reply:
It was suggested in the initial PR because of the `writer.deleteRecords` was implemented (for all partitions in config)
https://github.com/apache/kafka/pull/18014#discussion_r1871352116

However, the code re-written such that `writer.deleteRecords` would get called only for active shards and this is no longer the case. We can move that back to `error`. 

Great catch!